src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -0,0 +1,852 @@
+        ? this.config.noResultsMessage
+        : 'No results found';
+
+      tableBodyRows = (
+        <div class={DATA_TABLE_CLASSES.EMPTY}>
+          <div>{noResultsMessage}</div>
+        </div>
+      );
+    }
+
+    return <div class={DATA_TABLE_CLASSES.BODY}>{tableBodyRows}</div>;
+  }
+
+  pagination() {
+    const results = this.data.body.length,
+      pages =
+        this.resultsPerPage === this.data.body.length
+          ? 1
+          : Math.floor(results / this.resultsPerPage) + 1;
+
+    return (
+      <Pagination
+        ref="pagination"
+        compact={
+          this.config.style.portal || this.config.pagination.compact
+        }
+        firstLast={this.config.pagination.compact}
+        currentPage={this.currentPage}
+        pages={pages}
+        results={results}
+        pageSize={!this.config.style.portal}
+        pageJumper={this.config.pagination.pageJumper}
+      />
+    );
+  }
+
+  checkSelectAllCheckbox() {
+    const selectAllCheckbox = document.querySelector(
+      `.${DATA_TABLE_CLASSES.HEAD} .${DATA_TABLE_CLASSES.CELL} input`
+    ) as HTMLInputElement;
+
+    if (selectAllCheckbox) {
+      const isSelected = (row: DataTableRow) =>
+        this.selectedRows.includes(`row-${this._dataTableNumber}-${row.id}`);
+
+      selectAllCheckbox.checked = !!this.slicedData.every(isSelected);
+    }
+  }
+
+  sliceData(data: Record<string, any>) {
+    if (data.length > this.resultsPerPage) {
+      const arrayStartIndex = (this.currentPage - 1) * this.resultsPerPage,
+        arrayEndIndex =
+          (this.currentPage - 1) * this.resultsPerPage + this.resultsPerPage;
+
+      return data.slice(arrayStartIndex, arrayEndIndex);
+    }
+  }
+
+  detectScreenBreakpoint() {
+    const currentScreenSize = window.innerWidth;
+    let screenBreakpoint: DataTableScreenBreakpoints = 'xl';
+
+    switch (true) {
+      case currentScreenSize > SCREEN_BREAKPOINTS.LG:
+        screenBreakpoint = 'xl';
+        break;
+      case currentScreenSize > SCREEN_BREAKPOINTS.MD + 1 && currentScreenSize <= SCREEN_BREAKPOINTS.LG:
+        screenBreakpoint = 'lg';
+        break;
+      case currentScreenSize > SCREEN_BREAKPOINTS.SM + 1 && currentScreenSize <= SCREEN_BREAKPOINTS.MD:
+        screenBreakpoint = 'md';
+        break;
+      case currentScreenSize > SCREEN_BREAKPOINTS.XS + 1 && currentScreenSize <= SCREEN_BREAKPOINTS.SM:
+        screenBreakpoint = 'sm';
+        break;
+      case currentScreenSize < SCREEN_BREAKPOINTS.XS:
+        screenBreakpoint = 'xs';
+        break;
+    }
+
+    if (this._currentScreenBreakpoint !== screenBreakpoint) {
+      this._currentScreenBreakpoint = screenBreakpoint;
+      this.$forceUpdate();
+    }
+  }
+
+  created() {
+    this._expandable = !!this.data.body.find(
+      (row: { nestedContent: any }) => row.nestedContent
+    );
+    this.slicedData = this.sliceData(this.data.body);
+    this._dataTableNumber = `dt-${dataTableNumber}`;
+    dataTableNumber+=1;
```

Comment:
Please, move this to the constructor.

Reply:
![Screenshot 2021-02-05 at 18 19 53](https://user-images.githubusercontent.com/25136131/107067180-35a05500-67df-11eb-9ea6-00610885dab5.png)
When moving it to `constructor`, I encounter other issues. Also, looks like according to the Vue documentation `created() {}` is more appropriate

Reply:
@janibolkvadze Got it. In that case, the ! (exclamation mark) in the property definition is ok.